### PR TITLE
Build Mode Preview + better help text

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
Gives `Adv Build Mode` and `Fill` an item preview, and puts the help text for build mode modes into an examine block.

https://github.com/tgstation/tgstation/assets/66640614/6751c721-6ddd-4761-8311-fb002ea905ac

image

image

## Why It's Good For The Game
It's good to see what you're doing. 

## Changelog
:cl: Tattle
admin: build mode help text is in an examine block
admin: adv build mode and fill now have item previews
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
YES PLEASE

File: code/modules/buildmode/buildmode.dm [hunk 1]
```diff
@@ -22,6 +22,8 @@
 	// dirswitch UI
 	var/atom/movable/screen/buildmode/bdir/dirbutton
 	var/list/dirswitch_buttons = list()
+	// item preview for selected item
```
Comment (san7890) [inline code/modules/buildmode/buildmode.dm:25]:
```suggestion
	/// item preview for selected item
```

dmdoc is better, wish the rest of the datum had it but at the very least the new stuff should

File: code/modules/buildmode/buildmode.dm [hunk 2]
```diff
@@ -126,10 +129,42 @@
...
+	preview_overlay.transform = preview_overlay.transform.Scale(scale)
+	preview_overlay.appearance_flags |= TILE_BOUND
+	preview_overlay.layer = FLOAT_LAYER
+	preview_overlay.plane = FLOAT_PLANE
+	preview.add_overlay(preview_overlay)
+
+	holder.screen += preview
+
+/datum/buildmode/proc/close_preview()
+	if(!preview)
```
Comment (san7890) [inline code/modules/buildmode/buildmode.dm:157]:
```suggestion
	if(isnull(preview))
```

`isnull()` is more explicit here, and like 0.001 CPU cycles faster. nice.

File: code/modules/buildmode/buildmode.dm [hunk 3]
```diff
@@ -126,10 +129,42 @@
...
+		return
+	holder.screen -= preview
+	qdel(preview)
+	preview = null
```
Comment (san7890) [inline code/modules/buildmode/buildmode.dm:161]:
```suggestion
	QDEL_NULL(preview)
```

same thing, just in a neater macro

**Review by san7890 — CHANGES_REQUESTED** [review]

**Pull request merged**